### PR TITLE
The next installment of the reforms of the Storage Manager

### PR DESCRIPTION
**Pull request by jamiefaye**
In which we split the StorageManager.cpp into 3 files. We also are slowly eliminating "globalism" by allowing multiple Serializers and Deserializers to co-exist. The writeJsonFlag controls which file format we save in. RIght now it is bool const = false which may allow some of the code to be stripped away.

### CODE REVIEW

**Comment by seangoodvibes** [comment]
First comment: 

I see use of both f_close and closeFile. I think we should be consistent and use one or the other and not both?

File: src/deluge/gui/ui/load/load_song_ui.cpp [hunk 1]
```diff
@@ -284,7 +284,8 @@ void LoadSongUI::performLoad(StorageManager& bdsm) {
 
 someError:
 		display->displayError(error);
-		storageManager.closeFile();
+		f_close(&smDeserializer.readFIL);
```
Comment (m-m-adams) [inline src/deluge/gui/ui/load/load_song_ui.cpp:287]: This feels like it should still be done through the storage manager

File: src/deluge/gui/ui/save/save_song_ui.cpp [hunk 2]
```diff
@@ -237,7 +237,7 @@ bool SaveSongUI::performSave(StorageManager& bdsm, bool mayOverwrite) {
 				// converted
 
 				// Open file to read
-				FRESULT result = f_open(&fileSystemStuff.currentFile, sourceFilePath, FA_READ);
+				FRESULT result = f_open(&smSerializer.writeFIL, sourceFilePath, FA_READ);
```
Comment (m-m-adams) [inline src/deluge/gui/ui/save/save_song_ui.cpp:240]: Same here - I think we should keep the actual file system calls centralized

File: src/deluge/gui/ui/load/load_song_ui.cpp (hunk 1 above)
Comment (jamiefaye) [reply src/deluge/gui/ui/load/load_song_ui.cpp]: I moved file closing into the FileReader and FileWriter classes.

File: src/deluge/gui/ui/save/save_song_ui.cpp (hunk 2 above)
Comment (jamiefaye) [reply src/deluge/gui/ui/save/save_song_ui.cpp]: Also handled by above change.

**Pull request merged**